src/components/Chat/InputMessageV2.tsx

```diff
@@ -308,20 +318,17 @@ const InputMessageV2 = ({
 
   const onOpenFile = async () => {
     setIsLoadingUploadImageFile(true);
-    const {success} = await requestExternalStoragePermission();
-    if (success) {
-      DocumentPicker.pickSingle({
-        presentationStyle: 'fullScreen'
-      }).then((pickerResult) => {
+    DocumentPicker.pickSingle({
+      presentationStyle: 'fullScreen'
+    })
+      .then((pickerResult) => {
+        setIsLoadingUploadImageFile(false);
         refAttachment.current.close();
         handleFile(pickerResult);
+      })
+      .catch(() => {
+        setIsLoadingUploadImageFile(false);
       });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Again, the new code is missing a permission check before opening the file picker. It's important to ensure that the app has the necessary permissions before attempting to access the file system.

```diff
+    const {success} = await requestExternalStoragePermission();
+    if (success) {
         DocumentPicker.pickSingle({
           presentationStyle: 'fullScreen'
         })
         .then((pickerResult) => {
           setIsLoadingUploadImageFile(false);
           // rest of the code...
         })
         .catch(() => {
           setIsLoadingUploadImageFile(false);
         });
+    } else {
+      setIsLoadingUploadImageFile(false);
+      openAlertPermission(
+        'We’re not able to access your document library, please adjust your permission settings for BetterSocial.'
+      );
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->